Updates for js2.2.3

DennisDawson opened the pull request:
Update Quickstart tutorial for changes in language, add Broker Sale lesson.

DennisDawson committed:
Updates for 2.2.3

Remove section on using the account faucet from the docs; add the Destination fields.

DennisDawson committed:
Add Destination and revise code

DennisDawson committed:
Update with Destination field support

DennisDawson committed:
Update to add transfer fee

DennisDawson committed:
Update Transfer Fee description.

DennisDawson committed:
Fix format of tfTransferable

DennisDawson committed:
Remove odd bold period

DennisDawson committed:
Updates for 2.2.3

Add expiration date, fix indents.

DennisDawson committed:
Remove tabs.

DennisDawson committed:
WIP broker-sale.md

DennisDawson committed:
WIP

DennisDawson committed:
Remove unnecessary code.

No reason to get the wallet when getting the lists of offers.

DennisDawson committed:
Complete first draft

DennisDawson committed:
Remove default amount values

DennisDawson committed:
Update screenshots

DennisDawson committed:
Update screenshot and UI info

DennisDawson committed:
Add screenshots.

DennisDawson committed:
Update to NFTokenBrokerFee

DennisDawson committed:
Update alt text

DennisDawson committed:
Update and add screenshots

DennisDawson committed:
Add Broker Sale to list of lessons

DennisDawson committed:
Update Prev/Next

DennisDawson committed:
Add Broker NFT Sale

DennisDawson committed:
Merge branch 'master' into Updates_for_JS2.2.3

DennisDawson committed:
Updated code samples for 2.2.3

DennisDawson committed:
Add line to enable list indent

DennisDawson committed:
Separate Broker example from basic 4 lessons

Planning for the future, the requirement of 3 accounts on screen is specific to brokered sales. Other examples that do not require 3 accounts can be built off of the earlier examples. The brokered example is not necessarily in line with the future trajectory of the tutorial examples.

DennisDawson committed:
Remove enumeration from link to broker-sale

In consideration of future lessons, brokering a sale is not necessarily in line with the overall trajectory of the tutorials.

DennisDawson committed:
Remove arrow character

DennisDawson committed:
Fix error on Transfer Fee

If non-transferable, set the value to 0, not 1.

DennisDawson committed:
Ugh - remove WIP js file

mDuo13 commented on content/tutorials/quickstart/create-accounts-send-xrp.md:
```diff
@@ -243,7 +228,7 @@ Otherwise, populate the operational account fields.
 
 
-Insert the seed values for both accounts as they are created to the **Seeds** field as a convenience. You can copy the values and store them offline, then paste them into the **Seeds** field to retrieve the accounts with the `getAccountsFromSeeds()` function.
+Insert the seed values for both accounts as they are created to the **Seeds** field as a convenience. You can copy the values and store them offline. when you reload this form or another in this tutorial, copy and paste them into the **Seeds** field to retrieve the accounts with the `getAccountsFromSeeds()` function.
```
Looks like a missing capital letter at least. Not sure if there's a bigger typo going on here.

```suggestion
Insert the seed values for both accounts as they are created to the **Seeds** field as a convenience. You can copy the values and store them offline. When you reload this form or another in this tutorial, copy and paste them into the **Seeds** field to retrieve the accounts with the `getAccountsFromSeeds()` function.
```

mDuo13 commented on content/tutorials/quickstart/create-accounts-send-xrp.md:
```diff
@@ -539,18 +524,24 @@ Create a standard HTML form to send transactions and requests, then display the
     <h1>Token Test Harness</h1>
     <form id="theForm">
-      Choose your network:  
+      Choose your ledger instance:  
```
I suggest rolling this change back. The term "network" is more consistent with the usage across our other docs, and "ledger instance" is likely to be interpreted to mean an instance of the ledger data type.

DennisDawson replied:
Will do in the next release.

mDuo13 commented on content/tutorials/quickstart/create-trustline-send-currency.md:
```diff
@@ -53,7 +53,8 @@ To create a trustline between accounts:
 4. Enter the maximum transfer limit in the **Amount** field.
-5. Click **Create Trustline**.
+5. Enter the destination account value in the **Destination** field.
+6. Click **Create Trustline**.
```
Per our style guide, "trust line" is two words. (The corresponding UI element should also change.)

DennisDawson replied:
Agreed, but will do in a future release.

mDuo13 commented on content/tutorials/quickstart/mint-and-burn-nftokens.md:
```diff
@@ -5,7 +5,8 @@ blurb: Quickstart step 3, mint and burn NFTokens.
   - Tokens
-  - Non-fungible tokens, NFTs
+  - Non-fungible tokens
+  - NFTs
```
Changing the label to two labels requires changes to the CSS (to ensure both labels have appropriate colors) and to the dactyl config file to make sure both labels have landing pages.

To get this PR out the door ASAP, I suggest rolling this change back (in all places where it appears). We can address the issue more thoroughly in a follow-up.

```suggestion
  - Non-fungible tokens, NFTs
```

mDuo13 commented on content/tutorials/quickstart/mint-and-burn-nftokens.md:
```diff
@@ -20,7 +21,7 @@ This example shows how to:
 
 
-![Test harness with mint NFToken fields](img/quickstart13.png)
+![Test harness with mint NFToken fields](img/quickstart8.png)
```
It might be more effective to have descriptive rather than numbered file names for these images so that you don't have to renumber them every time you update the tutorial. Just a suggestion, no action needed on this now.

mDuo13 commented on content/tutorials/quickstart/mint-and-burn-nftokens.md:
```diff
@@ -122,15 +114,15 @@ Connect to the ledger and get the account wallets.
+  document.getElementById('standbyResultField').value = results
+  let net = getNet()
+  const standby_wallet = xrpl.Wallet.fromSeed(standbySeedField.value)
+  const client = new xrpl.Client(net)
```
The blurb on line 111 above says "account wallets" but this code only gets one wallet now. Not sure which one should be fixed.

mDuo13 commented on content/tutorials/quickstart/mint-and-burn-nftokens.md:
````diff
@@ -219,51 +220,51 @@ Disconnect from the ledger.
 Connect to the ledger and get the account wallet.
 
 
 ```
````
Another suggestion not important immediately: these code samples are a good use case for the `include_code` macro. That way you don't have to update the code in two places when changes are needed.

Also, syntax highlighting will work more reliably if you specify the language in the opening fence, e.g. \`\`\`js instead of just ```

mDuo13 commented on content/tutorials/quickstart/transfer-nftokens.md:
```diff
@@ -38,22 +38,14 @@ You can download the [Quickstart Samples](https://github.com/XRPLF/xrpl-dev-port
 
 1. Open `4.transfer-nftokens.html` in a browser.
-2. Get test accounts.
+2. Choose **XLS20-NFT** as your ledger instance.
```
Same comment as above re "ledger instance" vs "network"

mDuo13 commented on content/tutorials/quickstart/transfer-nftokens.md:
````diff
@@ -175,34 +169,59 @@ Connect to the ledger and get the wallet accounts.
+  document.getElementById('standbyResultField').value = results
+```
 
+Compute the Expiration Date, if present. The expiration date represents the number of seconds after the Ripple Epoch that the offer should expire. Subtract the Ripple Epoch from the current system date divided by 1,000, and add the number of days (or fractions of a day) multiplied by 86400 to get the correct expiration date.
````
This conversion is confusing (I didn't know that doing a division operation on a `Date()` object in JS implicitly converts it to milliseconds first) and a lot of this math could be simplified by using functions that are built into xrpl.js (see my comment on the actual code below).

mDuo13 commented on content/tutorials/quickstart/transfer-nftokens.md:
```diff
@@ -175,34 +169,59 @@ Connect to the ledger and get the wallet accounts.
+    var secondsInDay = 86400
+    var rippleEpoch = 946684800
+    var currentTimestamp = (new Date() / 1000) - rippleEpoch
+    expirationDate = parseInt(Math.floor(currentTimestamp + (days*secondsInDay)))
```
Rather than figuring out tricky format conversions on their own, we should encourage users to rely on the conversion functions that are built into xrpl.js. For example:

```suggestion
    let d = new Date()
    d.setDate(d.getDate() + parseInt(days))
    var expirationDate = xrpl.isoTimeToRippleTime(d)
```

(The difference between `let` and `var` is a separate thing, which is probably worth discussing at some point, but not necessary to fix at this time)

mDuo13 commented on content/tutorials/quickstart/broker-sale.md:
```diff
@@ -0,0 +1,1053 @@
+blurb: Broker a sale between a sell offer and a buy offer.
+labels:
+  - Accounts
+  - Quickstart
+  - Broker
```
At some point we should add landing pages and colors for these labels (they'll be unclickable gray labels as is). I can push a commit that does that later though.

mDuo13 reviewed (approved):
The labels stuff won't work quite as desired here, and there are some areas where I suggested some longer-term or follow-up improvements plus at least one typo. But nothing major that should block the updates from going on.

DennisDawson committed:
more explicit description of use of seeds

Co-authored-by: Rome Reginelli <rome@ripple.com>

DennisDawson committed:
revert NFT label

Not a big deal, a mistake on my part in the original. Ideally, this will be more flexible in the future. I didn't realize these were hard-coded.

Co-authored-by: Rome Reginelli <rome@ripple.com>

amarantha-k commented on content/tutorials/quickstart/create-accounts-send-xrp.md:
```diff
@@ -539,18 +524,24 @@ Create a standard HTML form to send transactions and requests, then display the
     <h1>Token Test Harness</h1>
     <form id="theForm">
-      Choose your network:  
+      Choose your ledger instance:  
```
```suggestion
      Choose your network:  
```

amarantha-k commented on content/tutorials/quickstart/create-trustline-send-currency.md:
```diff
@@ -53,7 +53,8 @@ To create a trustline between accounts:
 4. Enter the maximum transfer limit in the **Amount** field.
-5. Click **Create Trustline**.
+5. Enter the destination account value in the **Destination** field.
+6. Click **Create Trustline**.
```
```suggestion
6. Click **Create Trust line**.
```

DennisDawson committed:
Much better date conversion.

Co-authored-by: Rome Reginelli <rome@ripple.com>

amarantha-k commented on content/tutorials/quickstart/transfer-nftokens.md:
````diff
@@ -175,34 +169,59 @@ Connect to the ledger and get the wallet accounts.
+  document.getElementById('standbyResultField').value = results
+```
 
+Compute the Expiration Date, if present. The expiration date represents the number of seconds after the Ripple Epoch that the offer should expire. Subtract the Ripple Epoch from the current system date divided by 1,000, and add the number of days (or fractions of a day) multiplied by 86400 to get the correct expiration date.
````
```suggestion
Compute the Expiration Date, if present. The expiration date represents the number of seconds after the Ripple Epoch when the offer should expire. Subtract the Ripple Epoch from the current system date divided by 1,000, and add the number of days (or fractions of a day) multiplied by 86400 to get the correct expiration date.
```

DennisDawson replied:
Not sure I agree with this one, so I'll let it go rather than risk another edit.

DennisDawson committed:
Update description of expiry date

amarantha-k reviewed (approved):
Other than some suggestions added in the review, lgtm

DennisDawson committed:
update reciprocal expiry date

DennisDawson committed:
update reciprocal expiry

DennisDawson committed:
Fix trust line in descriptions

Leaving it in the UI for now, due to extensive changes required to get it right there.

DennisDawson committed:
Remove redundant comment

DennisDawson committed:
Use Rome's improved expiry date formula

DennisDawson committed:
Fix trust line

DennisDawson committed:
fix trust line in footer

DennisDawson committed:
fix trust line in footer

DennisDawson commented:
Merging, with future improvements noted.
